product_logistics_uom/models/product_template.py

```diff
@@ -0,0 +1,161 @@
+        default=lambda self: self._get_weight_uom_id_from_ir_config_parameter(),
+    )
+
+    weight_uom_name = fields.Char(
+        string="Weight unit of measure label",
+        related="weight_uom_id.name",
+        readonly=True,
+    )
+
+    show_volume_uom_warning = fields.Boolean(
+        help="Technical field used to warn the user to change the volume"
+        "uom since the value for product_volume is too small and has been"
+        "rounded.",
+        compute="_compute_show_volume_uom_warning",
+    )
+
+    show_weight_uom_warning = fields.Boolean(
+        help="Technical field used to warn the user to change the weight"
+        "uom since the value for product_weight is too small and has been"
+        "rounded.",
+        compute="_compute_show_weight_uom_warning",
+    )
+
+    @api.model
+    def _get_volume_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_volume_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_volume_uom_id_from_ir_config_parameter()
+
+    @api.model
+    def _get_weight_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_weight_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_weight_uom_id_from_ir_config_parameter()
+
+    @api.model
+    def _get_length_uom_id_from_ir_config_parameter(self):
+        get_param = self.env["ir.config_parameter"].sudo().get_param
+        default_uom = get_param("product_default_length_uom_id")
+        if default_uom:
+            return self.env["uom.uom"].browse(int(default_uom))
+        else:
+            return super()._get_length_uom_id_from_ir_config_parameter()
+
+    @api.depends(
+        "product_variant_ids",
+        "product_variant_ids.product_volume",
+        "volume",
+        "volume_uom_id",
+    )
+    def _compute_product_volume(self):
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        for template in unique_variants:
+            template.product_volume = template.product_variant_ids.product_volume
+        for template in self - unique_variants:
+            template.product_volume = 0.0
+
+    def _inverse_product_volume(self):
+        for template in self:
+            if len(template.product_variant_ids) == 1:
+                template.product_variant_ids.product_volume = template.product_volume
+
+    @api.depends("weight", "weight_uom_id")
+    def _compute_product_weight(self):
+        unique_variants = self.filtered(
+            lambda template: len(template.product_variant_ids) == 1
+        )
+        for template in unique_variants:
```

Comment:
I realize that this function uses 2 loops to process all the records in this model, you might consider reducing it to a single loop processing.
something like:
```
for template in self:
        template.product_weight = (
            template.product_variant_ids.product_weight
            if template.product_variant_count == 1
            else 0.0
        )
```

Reply:
It can also be applied to functions that use the two loops above. suggestions only, no mandatory changes required